### PR TITLE
Fix adding headers to config path

### PR DESCRIPTION
**Pull request by venigreat**
In this [PR](https://github.com/avito-tech/Emcee/pull/26) I added the ability to pass headers along with the path for the server configuration. 
A bug is fixed here - headers were not sent to the queue builder

### CODE REVIEW

File: Sources/ResourceLocation/ResourceLocation.swift [hunk 1]
```diff
@@ -116,8 +116,17 @@ public enum ResourceLocation: Hashable, CustomStringConvertible, Codable {
...
+            guard let headers = headers,
+                  headers.count > 0 else {
+                return url.absoluteString
+            }
+            let params: [String: Encodable] = [
+                "url": url.absoluteString,
+                "headers": headers
+            ]
+            let data = try! JSONSerialization.data(withJSONObject: params, options: .fragmentsAllowed)
+            return String(data: data, encoding: .utf8) ?? ""
```
Comment (beefon) [inline Sources/ResourceLocation/ResourceLocation.swift:129]: `try!` may cause a potential crash. Could you please make it `func stringValue() throws -> String` instead.

I've checked, there are not too many places where this string is used. Should be fairly straightforward to change property to a throwing method call.

Comment (venigreat) [reply Sources/ResourceLocation/ResourceLocation.swift:129]: change var to throwable func

File: Sources/ResourceLocation/ResourceLocation.swift [hunk 2]
```diff
@@ -112,12 +112,21 @@ public enum ResourceLocation: Hashable, CustomStringConvertible, Codable {
...
+            guard let headers = headers,
+                  headers.count > 0 else {
+                return url.absoluteString
+            }
+            let params: [String: Encodable] = [
+                "url": url.absoluteString,
+                "headers": headers
+            ]
+            let data = try JSONSerialization.data(withJSONObject: params, options: .fragmentsAllowed)
+            return String(data: data, encoding: .utf8) ?? ""
```
[the remark this answers is not in the document]
Comment (venigreat) [reply Sources/ResourceLocation/ResourceLocation.swift]: Fixed

File: Sources/TestDiscovery/TestDiscoveryQuerierImpl.swift [hunk 3]
```diff
@@ -197,8 +197,8 @@ public final class TestDiscoveryQuerierImpl: TestDiscoveryQuerier {
         testCount: Int,
         configuration: TestDiscoveryConfiguration,
         specificMetricRecorder: SpecificMetricRecorder
-    ) {
-        let testBundleName = configuration.xcTestBundleLocation.resourceLocation.stringValue.lastPathComponent
+    ) throws {
+        let testBundleName = try configuration.xcTestBundleLocation.resourceLocation.stringValue().lastPathComponent
```
[the remark this answers is not in the document]
Comment (venigreat) [reply Sources/TestDiscovery/TestDiscoveryQuerierImpl.swift]: Fixed

**Review by beefon — APPROVED** [review]

**Pull request merged**